tests/test_encode_decode.py

```diff
@@ -16,7 +17,19 @@ class TestBinASCII(unittest.TestCase):
 
 class TestColorsys(unittest.TestCase):
     # TODO: https://docs.python.org/3/library/colorsys.html
-    pass
+    @given(
+        r=st.floats(0.0, 1.0),
```

Comment:
I think it might be better to do this as a tuple, possibly a new strategy for it (since the strategy for rgb and hsv will be identical), so either:

```python
@given(
    rgb = st.tuples(st.floats(0.0, 1.0),
                    st.floats(0.0, 1.0),
                    st.floats(0.0, 1.0))
)
```

Or, if you want to do less repetition of the input arguments:

```python
rgb = st.tuples(*((st.floats(0.0, 1.0),) * 3))
``` 

You could probably refactor this as:

```python
color_element_st = st.floats(0.0, 1.0)
color_coordinate_st = st.tuples(*((color_element_st,) * 3))

...

    @given(rgb=color_coordinate_st)
    def test_rgb_hsv_round_trip(self, rgb):
        ...

    @given(hsv=color_coordinate_st)
    def test_hsv_rgb_round_trip(self, hsv):
        ...
```

Reply:
I'd stick to the first option you list, and keep things nice and explicit for now.  Separate arguments for each would also be a reasonable decision, because the range for `YIQ` colors is *not* `[0, 1]`...